enumerable.rb

```diff
@@ -0,0 +1,142 @@
+module Enumerable
+  # 1. my_each Method-----------------------------
+
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = if instance_of?(Range)
+            to_a
+          else
+            instance_of?(Hash) ? flatten : self
+          end
+
+    arr.length.times do |i|
+      yield to_a[i]
+    end
+
+    self
+  end
+
+  # 2. my_each_with_index Method-----------------------------
+
+  def my_each_with_index
```

Comment:
You technically give the right answer but you are giving the wrong output. It's an array

Reply:
![Screenshot from 2021-01-08 11-47-04](https://user-images.githubusercontent.com/31193823/104034471-4b523880-51a7-11eb-9fdc-824ec7457892.png)

Reply:
![Screenshot from 2021-01-08 11-47-47](https://user-images.githubusercontent.com/31193823/104034524-5dcc7200-51a7-11eb-8565-8df5aacae35e.png)